dashboard/app/models/user.rb

```diff
@@ -2075,6 +2077,32 @@ def destroy
     end
   end
 
+  # Called before_destroy.
+  # Soft-deletes any projects and other channel-backed progress belonging to
+  # this user.  Unfeatures any featured projects belonging to this user.
+  private def soft_delete_channels
+    return unless user_storage_id
+
+    channel_ids = PEGASUS_DB[:storage_apps].
+      where(storage_id: user_storage_id).
+      map(:id)
+
+    # Unfeature any featured projects owned by the user
+    FeaturedProject.
+      where(storage_app_id: channel_ids).
+      update_all(unfeatured_at: Time.now)
+
+    # Soft-delete all of the user's channels
+    PEGASUS_DB[:storage_apps].
+      where(id: channel_ids).
+      update(state: 'deleted')
```

Comment:
might later prove useful to also set `updated_at: Time.now`. If you do, you should probably also chain `.exclude(state: 'deleted')` before the `update`.